### ISSUE TITLE
Previews for .ods cause spiraling pandoc memory leak

### ISSUE THREAD

**Issue by shervinsahba** [ranger/ranger#2518]
#### Runtime Environment
- Operating system and version: 5.15.6-2-MANJARO 
- Terminal emulator and version: kitty 0.23.1 
- Python version: Python version: 3.9.9 (main, Nov 20 2021, 21:30:06) [GCC 11.1.0]
- Ranger version/commit: aur/ranger-git 1.9.3.26.g4bb83b37-1 (+146 0.04) (Installed: 1.9.3.387.g7e6a0aa6-1)
- Locale: en_US.UTF-8

#### Current Behavior

Certain file previews seem to fail and then turn into memory leaks. On my system, highlighting `.ods` files calls `pandoc` but doesn't preview. Instead, the pandoc task operates in the background and the memory consumption spirals ever upwards until all system memory is used. Quitting ranger does not terminate these pandoc calls.

pandoc 2.16.1 (compiled with pandoc-types 1.22.1, texmath 0.12.3.2, skylighting 0.12.1, citeproc 0.6, ipynb 0.1.0.2)

#### Expected Behavior
ods file previews should work, and memory spirals like this ought to be contained.

#### Steps to reproduce
1. highlight an ods file in ranger with previews enabled in the scope file
2. If the preview fails, check to see how pandoc is behaving.

Here's a gif. Not sure why it rendered choppily, but you can see the htop filtered results for pandoc and memory usage on the right.
![ranger_pandoc](https://user-images.githubusercontent.com/25020522/145699855-a8b6f250-5757-440f-8e8d-d78b7af31a02.gif)

**Comment by toonn** [comment]
Hmm, but we don't know the preview failed until pandoc finishes. This seems much more like a pandoc issue to me.

**Comment by shervinsahba** [comment]
Good point. Looks like pandoc [may not support ods](https://pandoc.org/MANUAL.html#options). At least I don't see the filetype mentioned on their docs nor in their GH issues. The terminal output
```
$ pandoc -s -t markdown -- ~/Downloads/blank.ods 
[WARNING] Could not deduce format from file extension .ods
  Defaulting to markdown
[WARNING] /home/shervin/Downloads/blank.ods is not UTF-8 encoded: falling back to latin1.
```
suggests pandoc doesn't know what to do with ods too.

Looking again at `scope.sh`, I noticed that I didn't have `odt2txt`. ([link to line](https://github.com/ranger/ranger/blob/master/ranger/data/scope.sh#L82)). The preview works fine with odt2txt installed! Without it, the ods option fell into pandoc and failed to render. I tested this for `odt` and `odp` filetypes as well. The latter has this issue too.

```
        ## OpenDocument
        odt|ods|odp|sxw)
            ## Preview as text conversion
            odt2txt "${FILE_PATH}" && exit 5
            ## Preview as markdown conversion 
            pandoc -s -t markdown -- "${FILE_PATH}" && exit 5  # <---- Does not work for ods, odp.
            exit 1;;
```

I'd suggest removing the pandoc line entirely or breaking up the cases.

**Comment by toonn** [comment]
PR is welcome, let's break up the cases.

**Comment by shervinsahba** [comment]
Merged fix #2520
